## EDA PROYECTO INSIGTH SEEKERS

In [1]:
# importar librerias
import pandas as pd

**Información general**

In [12]:
#apertura del archivo
df_info= pd.read_csv("hr_raw_data.csv", index_col = 0)

pd.set_option('display.max_rows', None)

# hacemos un ".head()" para mostrar las 5 primeras filas del dataframe
df_info

Age Attrition     BusinessTravel DailyRate  \
0               51        No                NaN    684,0$   
1               52        No                NaN    699,0$   
2               42        No      travel_rarely    532,0$   
3               47        No      travel_rarely    359,0$   
4               46        No                NaN   1319,0$   
5               48        No                NaN    117,0$   
6               59        No                NaN   1435,0$   
7               42        No      travel_rarely    635,0$   
8               41        No                NaN   1276,0$   
9               41        No  travel_frequently    840,0$   
10              41        No         non-travel    247,0$   
11              56        No      travel_rarely   1369,0$   
12              38        No                NaN    201,0$   
13              41       Yes      travel_rarely   1360,0$   
14              55        No                NaN    692,0$   
15              40        No      travel_rarely   1398,0$   
16              58       Yes      travel_rarely    286,0$   
17              46        No      travel_rarely   1402,0$   
18              35        No                NaN    819,0$   
19              40        No      travel_rarely    884,0$   
20              45        No         non-travel   1238,0$   
21              33        No                NaN    515,0$   
22              36        No                NaN   1223,0$   
23              47        No                NaN    202,0$   
24              36        No      travel_rarely    928,0$   
25              35        No                NaN    607,0$   
26              52       Yes                NaN    266,0$   
27              36        No                NaN    429,0$   
28              33        No      travel_rarely    589,0$   
29              34        No         non-travel      nan$   
30              41        No                NaN      nan$   
31     forty-seven        No                NaN   1180,0$   
32              53        No      travel_rarely   1282,0$   
33              35        No                NaN    776,0$   
34              34        No      travel_rarely    665,0$   
35              46        No      travel_rarely    526,0$   
36              43        No                NaN   1034,0$   
37              36        No                NaN   1403,0$   
38              60        No  travel_frequently   1499,0$   
39              40        No  travel_frequently    580,0$   
40              32        No      travel_rarely    859,0$   
41              41        No      travel_rarely    263,0$   
42              53        No                NaN   1376,0$   
43              36       Yes      travel_rarely    885,0$   
44              46        No                NaN   1003,0$   
45              38        No      travel_rarely   1321,0$   
46              51        No                NaN      nan$   
47              43        No                NaN    394,0$   
48              37        No                NaN   1372,0$   
49              32        No      travel_rarely      nan$   
50              38        No                NaN   1333,0$   
51              46        No                NaN    228,0$   
52              35       Yes      travel_rarely    737,0$   
53              43        No                NaN    823,0$   
54              56        No         non-travel    667,0$   
55              36        No         non-travel    301,0$   
56              38        No                NaN    573,0$   
57              49        No                NaN      nan$   
58              39        No      travel_rarely   1329,0$   
59              40        No                NaN    630,0$   
60              36        No                NaN      nan$   
61              37        No                NaN   1063,0$   
62              35        No                NaN   1017,0$   
63              33        No  travel_frequently   1296,0$   
64              50        No      travel_rarely   

In [7]:
# Cantidad de columnas y filas 

print(f"El número de filas que tenemos es {df_info.shape[0]}, y el número de columnas es {df_info.shape[1]}")

El número de filas que tenemos es 1614, y el número de columnas es 41


In [8]:
# TOTA REGISTROS = 1614

df_info.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1614 entries, 0 to 1613
Data columns (total 41 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Age                       1614 non-null   object 
 1   Attrition                 1614 non-null   object 
 2   BusinessTravel            842 non-null    object 
 3   DailyRate                 1614 non-null   object 
 4   Department                302 non-null    object 
 5   DistanceFromHome          1614 non-null   int64  
 6   Education                 1614 non-null   int64  
 7   EducationField            869 non-null    object 
 8   employeecount             1614 non-null   int64  
 9   employeenumber            1183 non-null   object 
 10  EnvironmentSatisfaction   1614 non-null   int64  
 11  Gender                    1614 non-null   int64  
 12  HourlyRate                1614 non-null   object 
 13  JobInvolvement            1614 non-null   int64  
 14  JobLevel     

`Anotaciones`

**Columnas con nulos**

- BusinessTravel : muchos nulos, casi la mitad
- Department :muchos nulos, mas de  la mitad
- EducationField : muchos nulos, casi la mitad
- employeenumber : algunos nulos
- MaritalStatus : muchos nulos, casi la mitad
- MonthlyIncome:  muchos nulos, casi la mitad
- Over18 : muchos nulos, mas de  la mitad
- OverTime: muchos nulos, casi la mitad
- PerformanceRating: algunos nulos
- StandardHours: muchos nulos, mas de  la mitad
- TOTALWORKINGYEARS : algunos nulos
- WORKLIFEBALANCE: algunos nulos
- YearsInCurrentRole: muchos nulos, mas de  la mitad
- SameAsMonthlyIncome : muchos nulos, casi la mitad
-  RoleDepartament : muchos nulos, mas de  la mitad
-  NUMBERCHILDREN : no hay datos

**Posibles cambios en tipo dato y limpiez**
 #   Column                       Tipo Dato                          Nulo                                               Limpieza
---  ------                    --------------                    --------------                         --------------------------------------- 
 0   Age                      object cambiar a int                  Non-null                             Los que estan string, cambiar a ordinal
 1   Attrition                 NO REQUIERE CAMBIOS                  Non-null                                     NO REQUIERE CAMBIOS 
 2   BusinessTravel            NO REQUIERE CAMBIOS              Nulos, casi mitad                             Igualar sintaxis(guiones)
 3   DailyRate                 object cambiar a float               Non-null                       Cambiar 'nan$' a número, agregar $ en columna
 4   Department                NO REQUIERE CAMBIOS              Nulos, mas de  la mitad                          NO REQUIERE CAMBIOS
 5   DistanceFromHome          NO REQUIERE CAMBIOS                  Non-null                   Todos en valores positivos.Agregar colum(km-mill)
 6   Education                 NO REQUIERE CAMBIOS                  Non-null                                         NO REQUIERE CAMBIOS 
 7   EducationField            NO REQUIERE CAMBIOS          Muchos nulos, casi la mitad                          NO REQUIERE CAMBIOS
 8   employeecount             NO REQUIERE CAMBIOS                  Non-null                                         NO REQUIERE CAMBIOS 
 9   employeenumber            object cambiar a int             algunos nulos                                    NO REQUIERE CAMBIOS 
 10  EnvironmentSatisfaction   NO REQUIERE CAMBIOS                  Non-null                                           NO REQUIERE CAMBIOS
 11  Gender (Genero)           int cambiar a object                 Non-null                                        NO REQUIERE CAMBIOS
 12  HourlyRate                object cambiar a float               Non-null           Not Available debe cambiarse- se puede calcular posterior
 13  JobInvolvement            NO REQUIERE CAMBIOS                  Non-null                                         NO REQUIERE CAMBIOS
 14  JobLevel                  NO REQUIERE CAMBIOS                  Non-null                            sintaxis con capitalize letter
 15  JobRole                   NO REQUIERE CAMBIOS                  Non-null                                    NO REQUIERE CAMBIOS
 16  JobSatisfaction           NO REQUIERE CAMBIOS                  Non-null                                   NO REQUIERE CAMBIOS
 17  MaritalStatus             NO REQUIERE CAMBIOS            muchos nulos, casi la mitad                   corregir sintaxis, capitalize
 18  MonthlyIncome             object cambiar a float      muchos nulos, casi la mitad                         agregar $ en columna
 19  MonthlyRate               int cambiar a float                  Non-null                                   agregar $ en columna
 20  NUMCOMPANIESWORKED        NO REQUIERE CAMBIOS                  Non-null                             cambiar columna en capitalize
 21  Over18                    NO REQUIERE CAMBIOS          muchos nulos, casi la mitad             se puede actualizar con age, cambiar Y-Yes
 22  OverTime                  NO REQUIERE CAMBIOS          muchos nulos, casi la mitad                 se pueden calcular los nulos
 23  PercentSalaryHike         1614 non-null   int64  
 24  PerformanceRating         1419 non-null   object 
 25  RelationshipSatisfaction  1614 non-null   int64  
 26  StandardHours             419 non-null    object 
 27  StockOptionLevel          1614 non-null   int64  
 28  TOTALWORKINGYEARS         1088 non-null   object 
 29  TrainingTimesLastYear     1614 non-null   int64  
 30  WORKLIFEBALANCE           1506 non-null   object 
 31  YearsAtCompany            1614 non-null   int64  
 32  YearsInCurrentRole        34 non-null     object 
 33  YearsSinceLastPromotion   1614 non-null   int64  
 34  YEARSWITHCURRMANAGER      1614 non-null   int64  
 35  SameAsMonthlyIncome       771 non-null    object 
 36  DateBirth                 1614 non-null   int64  
 37  Salary                    1614 non-null   object 
 38  RoleDepartament           302 non-null    object 
 39  NUMBERCHILDREN            0 non-null      float64
 40  RemoteWork                1614 non-null   object 